tools/src/main/java/org/apache/kafka/tools/GroupsCommand.java

```diff
@@ -0,0 +1,298 @@
+            } else if (protocolFilter.isPresent()) {
+                pass = protocol.equals(protocolFilter.get());
+            } else if (consumerGroupFilter) {
+                pass = protocol.equals("consumer") || protocol.isEmpty() || groupType.filter(gt -> gt == GroupType.CONSUMER).isPresent();
+            } else if (shareGroupFilter) {
+                pass = groupType.filter(gt -> gt == GroupType.SHARE).isPresent();
+            }
+            return pass;
+        }
+
+        @Override
+        public void close() throws Exception {
+            adminClient.close();
+        }
+    }
+
+    private static void printException(Throwable e) {
+        System.out.println("Error while executing groups command : " + e.getMessage());
+        LOG.error(Utils.stackTrace(e));
+    }
+
+    public static final class GroupsCommandOptions extends CommandDefaultOptions {
+        private final ArgumentAcceptingOptionSpec<String> bootstrapServerOpt;
+
+        private final ArgumentAcceptingOptionSpec<String> commandConfigOpt;
+
+        private final OptionSpecBuilder listOpt;
+
+        private final ArgumentAcceptingOptionSpec<String> groupTypeOpt;
+
+        private final ArgumentAcceptingOptionSpec<String> protocolOpt;
+
+        private final OptionSpecBuilder consumerOpt;
+
+        private final OptionSpecBuilder shareOpt;
+
+        public GroupsCommandOptions(String[] args) {
+            super(args);
+            bootstrapServerOpt = parser.accepts("bootstrap-server", "REQUIRED: The Kafka server to connect to.")
+                    .withRequiredArg()
+                    .describedAs("server to connect to")
+                    .ofType(String.class);
+            commandConfigOpt = parser.accepts("command-config", "Property file containing configs to be passed to Admin Client.")
+                    .withRequiredArg()
+                    .describedAs("command config property file")
+                    .ofType(String.class);
+
+            listOpt = parser.accepts("list", "List the groups.");
+
+            groupTypeOpt = parser.accepts("group-type", "Filter the groups based on group type. "
+                            + "Valid types are: 'classic', 'consumer' and 'share'.")
+                    .withRequiredArg()
+                    .describedAs("type")
+                    .ofType(String.class);
+
+            protocolOpt = parser.accepts("protocol", "Filter the groups based on protocol type.")
+                    .withRequiredArg()
+                    .describedAs("protocol")
+                    .ofType(String.class);
+
+            consumerOpt = parser.accepts("consumer", "Filter the groups to show all kinds of consumer groups, including classic and simple consumer groups. "
+                            + "This matches group type 'consumer', and group type 'classic' where the protocol type is 'consumer' or empty.");
+            shareOpt = parser.accepts("share", "Filter the groups to show share groups.");
```

Comment:
Does it equal to `group-type=share`?

Reply:
Yes, exactly. Share groups are modern groups.

Reply:
should we remove both `share` or `consumer` if they can be replaced by `group-type=share` and `group-type=consumer`?

Reply:
I think these options greatly improve the usability for common cases - just show me all of the consumer groups, and just show me all of the share groups.